### ISSUE TITLE
Add Link with hover and focus states to Story Promo

### ISSUE THREAD

**Issue by andrew-nowak** [BBC-archive/psammead#475]
**Is your feature request related to a problem? Please describe.**
The story promo component should change appearance when the user hovers or focuses on the component.

**Describe the solution you'd like**
![image](https://user-images.githubusercontent.com/10963046/56818257-27b65580-683f-11e9-83c2-ff7d79fdc156.png)
(See zeplin for other examples)
Add an underline to the headline. The hover state should be toggled when hovering anywhere on the component; not just the headline itself.

**Implementation details:**
The way that inline links work in simorgh requires the link component to be integrated with a `react-router-dom` link as seen [here](https://github.com/bbc/simorgh/blob/latest/src/app/containers/InlineLink/index.jsx#L12).

To do this, make the link component exportable like this

```
import StoryPromo, { Headline, Summary, Link } from '@bbc/psammead-story-promo';
```

Which can then be constructed like
```
<Link href="href">
   <StoryPromo
     image={Image}
     info={Info}
   />
</Link>
```

**Describe alternatives you've considered**
UX previously discussed using border-bottom instead of the font underline; this has been rejected and we should use the font underline.

**Testing notes**
Testing to investigate and attempt Cypress tests for states

Dev insight: Storybook and unit tests

**Additional context**

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/psammead/labels/Refinement%20Needed)

**Comment by greenc05** [comment]
Please look at the screen reader UX before working on this ticket, you will note that the link should not wrap all the elements in the promo and only be applied to what is semantically the link e.g. the heading.

Doing some screen reader testing, **we have better support if we nest the link inside the h3 tag** as follows (rather than having a link with a h3 nested inside):

```
<h3>
    <a href="my-link">Restoring a pharaoh's ancient boat</a>
</h3>
```

We can then use faux-block-link to create a large clickable area (or a similar CSS technique) https://github.com/bbc/grandstand/blob/master/docs/objects/faux-block-link.md

You may want to split this Issue into 2, one for adding the link HTML and one for implementing the whole clickable area....
